libnavigation-util/src/main/java/com/mapbox/navigation/utils/internal/LoggerProvider.kt

```diff
@@ -68,3 +105,20 @@ fun logW(msg: String, category: String? = null) {
 fun logE(msg: String, category: String? = null) {
     LoggerProvider.frontend.logE(msg, category)
 }
+
+/**
+ * @param category optional string to identify the source or category of the log message.
+ * @param lazyMsg is a lazy message to log. Isn't executed if current log level less verbose than Error.
+ * Noting that the category is appended to the log message to give extra context along with the `[nav-sdk]` parent category.
+ * As an example, this is how the logs would look like `E/Mapbox: [nav-sdk] [ConnectivityHandler] NetworkStatus=ReachableViaWiFi`.
+ */
+inline fun logE(category: String? = null, lazyMsg: () -> String) {
+    if (logLevel().atLeast(LoggingLevel.ERROR)) {
+        logE(lazyMsg(), category)
+    }
+}
+
+/**
+ * Should not be used directly. Added to support inline calls.
```

Comment:
How does it help with inlining?

Reply:
Clarified documentation. Does it make sense now?

Reply:
Aaah, I see. But why shouldn't it be used directly? It doesn't look like it can hurt anyone.

Reply:
It just feels like internal implementation detail and I can't keep it internal because of inline, so I decided to restrict usage by forbidding it in the comments  😅